src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -539,5 +611,75 @@ public void testTemplateArrayAccess() throws Exception {
 		//check that both @Parameter usage was replaced by appropriate parameter value
 		CtMethod<?> m2 = resultKlass.getMethod("method2");
 		assertEquals("java.lang.System.out.println(\"second\")", m2.getBody().getStatement(0).toString());
+		//check that substitution by missing value correctly produces empty expression
+		assertEquals("java.lang.System.out.println(null)", m2.getBody().getStatement(1).toString());
+	}
+
+	@Test
+	public void testStatementTemplateRootSubstitution() throws Exception {
+		//contract: the template engine supports substitution of root element
```

Comment:
SubstituteRootTemplate extends StatementTemplate, which internally calls
```java
CtStatement result = c.getMethod("statement").getBody().getStatements().get(0).clone();
```
where `result` is `block.S();` ... so it is substitution of the root element in the spoon model represented by `result`.